### DISCUSSION TITLE
Unable to set xonsh as default

### DISCUSSION THREAD

**Discussion by neodiksh** [xonsh/xonsh#4956] [Q&A] [answered]
Here are the outputs
~ 
❯ chsh -s /usr/bin/xonsh
chsh: your shell is not in /etc/shells, shell change denied: Permission denied

~ 
❯ chsh -l               
/bin/sh
/bin/bash
/usr/bin/git-shell
/bin/zsh
/usr/bin/zsh
/usr/bin/xonsh

~ 
❯ which xonsh
/usr/bin/xonsh

**Comment by gforsyth** [comment, marked as answer]
Hmm, that's weird.  Is it possible that the entry in `/etc/shells` has a trailing space in it?

**Reply by neodiksh** [reply]
don't know the but now its working when i have created a new USER,maybe some problem with my previous main user couldn't login pc as well
